qa/rpc-tests/test_framework/blocktools.py

```diff
@@ -58,20 +64,26 @@ def serialize_script_num(value):
         r.append(0x80 if neg else 0)
     elif neg:
         r[-1] |= 0x80
+    r.insert(0, len(r))
     return r
 
 # Create a coinbase transaction, assuming no miner fees.
 # If pubkey is passed in, the coinbase output will be a P2PK output;
 # otherwise an anyone-can-spend output.
-def create_coinbase(height, pubkey = None):
+def create_coinbase(height, pubkey = None, amount = 0):
     coinbase = CTransaction()
-    coinbase.vin.append(CTxIn(COutPoint(0, 0xffffffff), 
-                ser_string(serialize_script_num(height)), 0xffffffff))
+    # coinbase transaction scriptsigs must be at least 2 bytes
+    coinbase.vin.append(CTxIn(COutPoint(0, 0xffffffff),
+                serialize_script_num(height) + b'\x51', 0xffffffff))
     coinbaseoutput = CTxOut()
-    coinbaseoutput.nValue = 50 * COIN
-    halvings = int(height/150) # regtest
-    coinbaseoutput.nValue >>= halvings
-    if (pubkey != None):
+
+    coinbaseoutput.nValue.setToAmount(amount)
+    coinbaseoutput.nAsset.setToAsset(b'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa')
+
+    if amount == 0:
+        # zero utxo's must be made unspendable
```

Comment:
this comment refers to [this](https://github.com/ElementsProject/elements/blob/elements-0.14.1/src/validation.cpp#L1007) condition.